.github/workflows/release_pypi.yml

```diff
@@ -0,0 +1,204 @@
+name: Publish to PyPI
+
+on:
+  # Run on any tag
+  push:
+    tags:
+      - '**'
+  # manually trigger the workflow - for testing only
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build Distribution
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+      - name: Install pypa/build
+        run: >-
+          python3 -m pip install build --user
+      - name: Build a binary wheel and a source tarball
+        run: python3 -m build
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v4
+        with:
+          name: python-package-distributions
+          path: dist/
+
+  publish-to-pypi:
+    name: Publish to PyPI
+    # only publish to PyPI on tag pushes
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    needs:
+      - build
+    runs-on: ubuntu-latest
+    environment:
+      name: release_pypi
```

Comment:
does this need to match the file name? In the [docs](https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/), they put it as "pypi". 

If so, can you leave a comment saying it needs to match the file name?

Reply:
it's not the file name but rather the name of the [environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) :)